internal/pkg/s3/s3.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package s3 provides a client to retrieve Copilot S3 information.
+package s3
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/copilot-cli/internal/pkg/aws/resourcegroups"
+	"github.com/aws/copilot-cli/internal/pkg/aws/s3"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+)
+
+const (
+	bucketType = "s3:bucket"
+)
+
+type resourceGetter interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]*resourcegroups.Resource, error)
+}
+
+// Client retrieves Copilot S3 service information from AWS.
+type Client struct {
+	rgGetter resourceGetter
+}
+
+// New inits a new Client.
+func New(sess *session.Session) *Client {
```

Comment:
Rather than the session, should we aim for all packages except `cli` to pass down dependencies?

```
type ResourcesByTagsGetter interface {
   GetResourcesByTags(resourceType string, tags map[string]string) ([]*resourcegroups.Resource, error)
}

func New(rgapi ResourcesByTagsGetter) *Client
```

Reply:
I'm trying to be consistent with `internal/ecs` pkg. Maybe we can do a campaign to make this behavior consistent across the codebase (add it to the sprint planning task)?